build.sbt

```diff
@@ -0,0 +1,436 @@
+  scalacOptions ++= Seq("-deprecation", "-feature", "-unchecked"),
+  scalacOptions in (Compile, doc) ++= Seq("-skip-packages", ""),
+  scalacOptions in (Compile, doc) ++= Seq("-implicits", "-implicits-hide:."),
+  scalacOptions in (Compile, doc) ++= Seq("-groups"),
+  scalacOptions ++= Seq("-Xfatal-warnings"),
+  parallelExecution in Test := false, // hello, reflection sync!!
+  logBuffered := false,
+  triggeredMessage in ThisBuild := Watched.clearWhenTriggered
+)
+
+def computePreReleaseVersion(LibrarySeries: String): String = {
+  val preReleaseSuffix = {
+    import sys.process._
+    val stableSha = Try(git.stableSha()).toOption
+    val commitSubjects = Try(augmentString(shell.check_output("git log -10 --pretty=%s", cwd = ".")).lines.toList).getOrElse(Nil)
+    val prNumbers = commitSubjects.map(commitSubject => {
+      val Merge = "Merge pull request #(\\d+).*".r
+      val Squash = ".*\\(#(\\d+)\\)".r
+      commitSubject match {
+        case Merge(prNumber) => Some(prNumber)
+        case Squash(prNumber) => Some(prNumber)
+        case _ => None
+      }
+    })
+    val mostRecentPrNumber = prNumbers.flatMap(_.toList).headOption
+    (stableSha, prNumbers, mostRecentPrNumber) match {
+      case (Some(_), Some(prNumber) +: _, _) => prNumber
+      case (_, _, Some(prNumber)) => prNumber + "." + System.currentTimeMillis()
+      case _ => "unknown" + "." + System.currentTimeMillis()
+    }
+  }
+  LibrarySeries + "-" + preReleaseSuffix
+}
+
+// Pre-release versions go to bintray and should be published via `publish`.
+// This is the default behavior that you get without modifying the build.
+// The only exception is that we take extra care to not publish on pull request validation jobs in Drone.
+def shouldPublishToBintray: Boolean = {
+  if (!new File(sys.props("user.home") + "/.bintray/.credentials").exists) return false
+  if (sys.props("sbt.prohibit.publish") != null) return false
+  if (sys.env.contains("CI_PULL_REQUEST")) return false
+  LibraryVersion.contains("-")
+}
+
+// Release versions go to sonatype and then get synced to maven central.
+// They should be published via `publish-signed` and signed by someone from <developers>.
+// In order to publish a release version, change LibraryVersion to be equal to LibrarySeries.
+def shouldPublishToSonatype: Boolean = {
+  if (!secret.obtain("sonatype").isDefined) return false
+  if (sys.props("sbt.prohibit.publish") != null) return false
+  !LibraryVersion.contains("-")
+}
+
+lazy val publishableSettings = Def.settings(
+  sharedSettings,
+  bintrayOrganization := Some("scalameta"),
+  publishArtifact in Compile := true,
+  publishArtifact in Test := false,
+  {
+    val publishingStatus = {
+      if (shouldPublishToBintray) "publishing to Bintray"
+      else if (shouldPublishToSonatype) "publishing to Sonatype"
+      else "publishing disabled"
+    }
+    println(s"[info] Welcome to scala.meta $LibraryVersion ($publishingStatus)")
```

Comment:
Try using [sLog](http://www.scala-sbt.org/release/sxr/sbt/Keys.scala.html#sbt.Keys.sLog), but you'll have to be in the RHS of `publish in Compile`. Not sure if outside or inside Def.taskDyn though..